orchagent/bfdorch.cpp

```diff
@@ -0,0 +1,441 @@
+        {
+            if (session_type_map.find(value) == session_type_map.end())
+            {
+                SWSS_LOG_ERROR("Invalid BFD session type %s\n", value.c_str());
+                continue;
+            }
+            bfd_session_type = session_type_map.at(value);
+        }
+        else if (fvField(i) == "dst_mac")
+        {
+            dst_mac = MacAddress(value);
+            dst_mac_provided = true;
+        }
+        else
+            SWSS_LOG_ERROR("Unsupported BFD attribute %s\n", fvField(i).c_str());
+    }
+
+    if (!src_ip_provided)
+    {
+        SWSS_LOG_ERROR("Failed to create BFD session %s because source IP is not provided", key.c_str());
+        return true;
+    }
+
+    attr.id = SAI_BFD_SESSION_ATTR_TYPE;
+    attr.value.s32 = bfd_session_type;
+    attrs.emplace_back(attr);
+    fvVector.emplace_back("type", session_type_loopup.at(bfd_session_type));
+
+    attr.id = SAI_BFD_SESSION_ATTR_LOCAL_DISCRIMINATOR;
+    attr.value.u32 = bfd_gen_id();
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_UDP_SRC_PORT;
+    attr.value.u32 = bfd_src_port();
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_REMOTE_DISCRIMINATOR;
+    attr.value.u32 = 0;
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_BFD_ENCAPSULATION_TYPE;
+    attr.value.s32 = encapsulation_type;
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_IPHDR_VERSION;
+    attr.value.u8 = src_ip.isV4() ? 4 : 6;
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_SRC_IP_ADDRESS;
+    copy(attr.value.ipaddr, src_ip);
+    attrs.emplace_back(attr);
+    fvVector.emplace_back("local_addr", src_ip.to_string());
+
+    attr.id = SAI_BFD_SESSION_ATTR_DST_IP_ADDRESS;
+    copy(attr.value.ipaddr, peer_address);
+    attrs.emplace_back(attr);
+
+    attr.id = SAI_BFD_SESSION_ATTR_MIN_TX;
+    attr.value.u32 = tx_interval * BFD_SESSION_MILLISECOND_TO_MICROSECOND;
+    attrs.emplace_back(attr);
+    fvVector.emplace_back("tx_interval", to_string(tx_interval));
+
+    attr.id = SAI_BFD_SESSION_ATTR_MIN_RX;
+    attr.value.u32 = rx_interval * BFD_SESSION_MILLISECOND_TO_MICROSECOND;
+    attrs.emplace_back(attr);
+    fvVector.emplace_back("rx_interval", to_string(rx_interval));
+
+    attr.id = SAI_BFD_SESSION_ATTR_MULTIPLIER;
+    attr.value.u8 = multiplier;
+    attrs.emplace_back(attr);
+    fvVector.emplace_back("multiplier", to_string(multiplier));
+
+    if (multihop)
+    {
+        attr.id = SAI_BFD_SESSION_ATTR_MULTIHOP;
+        attr.value.booldata = true;
+        attrs.emplace_back(attr);
+        fvVector.emplace_back("multihop", "true");
+    }
+    else
+    {
+        fvVector.emplace_back("multihop", "false");
+    }
+
+    if (alias != "default")
+    {
+        Port port;
+        if (!gPortsOrch->getPort(alias, port))
+        {
+            SWSS_LOG_ERROR("Failed to locate port %s", alias.c_str());
+            return true;
```

Comment:
this should return `false `as a retry is expected here. port may get created later

Reply:
Updated as suggested.